org.envirocar.app/src/org/envirocar/app/views/settings/SettingsActivity.java

```diff
@@ -136,34 +140,27 @@ public void onDisplayPreferenceDialog(Preference preference) {
         }
 
         public void requestMicrophonePermission() {
-            String[] perms = new String[]{Manifest.permission.RECORD_AUDIO};
 
             // if microphone permission is not granted, request it
             if (!(requireContext().checkCallingOrSelfPermission(Manifest.permission.RECORD_AUDIO)
                     == PackageManager.PERMISSION_GRANTED)) {
-                requestPermissions(perms, RECORD_AUDIO_PERMISSION_REQ_CODE);
+                requestPermissionLauncher.launch(Manifest.permission.RECORD_AUDIO);
             }
         }
 
-        @Override
-        public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-            super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-            if (requestCode == RECORD_AUDIO_PERMISSION_REQ_CODE) {
-                if (grantResults.length > 0
-                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-
-                    //If user grants permission then show the snackbar
-                    Snackbar.make(requireView(), R.string.microphone_permission_granted, Snackbar.LENGTH_LONG).show();
-                } else {
-                    //If user denies permission then uncheck the checkbox back
-                    // and show the Snackbar to grant permission
-                    this.enableVoiceCommand.setChecked(false);
-
-                    // action opens app's general settings where user can grant microphone/any permission
-                    showGrantMicrophonePermission(requireView(), requireContext(), getActivity());
+        private ActivityResultLauncher<String> requestPermissionLauncher =
+                registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
+                    if (isGranted) {
+                        // Permission is granted. Continue the action or workflow in your
+                        // app.
+                        Snackbar.make(requireView(), R.string.microphone_permission_granted, Snackbar.LENGTH_LONG).show();
+                    } else {
 
-                }
-            }
-        }
+                        this.enableVoiceCommand.setChecked(false);
+
+                        // action opens app's general settings where user can grant microphone/any permission
+                        showGrantMicrophonePermission(requireView(), requireContext(), getActivity());
+                    }
+                });
```

Comment:
Personally, I prefer instantiating member objects inside the `onCreate()` method. Maybe, I will adjust it later. Since it works your way, there is no obstacle for merge this PR. But leave this comment here as a reminder.